{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reset env - import/install package"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 2 × 6 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>used</th><th scope=col>(Mb)</th><th scope=col>gc trigger</th><th scope=col>(Mb)</th><th scope=col>max used</th><th scope=col>(Mb)</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ncells</th><td> 595856</td><td>31.9</td><td>1271675</td><td>68</td><td> 595856</td><td>31.9</td></tr>\n",
       "\t<tr><th scope=row>Vcells</th><td>1066817</td><td> 8.2</td><td>8388608</td><td>64</td><td>1066817</td><td> 8.2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 2 × 6 of type dbl\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & used & (Mb) & gc trigger & (Mb) & max used & (Mb)\\\\\n",
       "\\hline\n",
       "\tNcells &  595856 & 31.9 & 1271675 & 68 &  595856 & 31.9\\\\\n",
       "\tVcells & 1066817 &  8.2 & 8388608 & 64 & 1066817 &  8.2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 2 × 6 of type dbl\n",
       "\n",
       "| <!--/--> | used | (Mb) | gc trigger | (Mb) | max used | (Mb) |\n",
       "|---|---|---|---|---|---|---|\n",
       "| Ncells |  595856 | 31.9 | 1271675 | 68 |  595856 | 31.9 |\n",
       "| Vcells | 1066817 |  8.2 | 8388608 | 64 | 1066817 |  8.2 |\n",
       "\n"
      ],
      "text/plain": [
       "       used    (Mb) gc trigger (Mb) max used (Mb)\n",
       "Ncells  595856 31.9 1271675    68    595856  31.9\n",
       "Vcells 1066817  8.2 8388608    64   1066817   8.2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# clear all elements from environment\n",
    "rm(list=ls())      #will clear all objects includes hidden objects.\n",
    "gc(reset = TRUE)   #free up memrory and report the memory usage."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocManager\n",
      "\n",
      "Warning message:\n",
      "\"package 'BiocManager' was built under R version 4.2.0\"\n",
      "Bioconductor version '3.14' is out-of-date; the current release version '3.15'\n",
      "  is available with R version '4.2'; see https://bioconductor.org/install\n",
      "\n",
      "Loading required package: GEOquery\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Setting options('download.file.method.GEOquery'='auto')\n",
      "\n",
      "Setting options('GEOquery.inmemory.gpl'=FALSE)\n",
      "\n",
      "Loading required package: minfi\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:grDevices':\n",
      "\n",
      "    windows\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: 'matrixStats'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:Biobase':\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'MatrixGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "The following object is masked from 'package:Biobase':\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: 'Biostrings'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: bumphunter\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "Warning message:\n",
      "\"package 'foreach' was built under R version 4.2.0\"\n",
      "Loading required package: iterators\n",
      "\n",
      "Warning message:\n",
      "\"package 'iterators' was built under R version 4.2.0\"\n",
      "Loading required package: parallel\n",
      "\n",
      "Loading required package: locfit\n",
      "\n",
      "locfit 1.5-9.4 \t 2020-03-24\n",
      "\n",
      "Loading required package: purrr\n",
      "\n",
      "\n",
      "Attaching package: 'purrr'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:locfit':\n",
      "\n",
      "    none\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:foreach':\n",
      "\n",
      "    accumulate, when\n",
      "\n",
      "\n",
      "The following object is masked from 'package:XVector':\n",
      "\n",
      "    compact\n",
      "\n",
      "\n",
      "The following object is masked from 'package:GenomicRanges':\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "The following object is masked from 'package:IRanges':\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "Loading required package: strex\n",
      "\n",
      "Loading required package: stringr\n",
      "\n",
      "Warning message:\n",
      "\"package 'stringr' was built under R version 4.2.0\"\n",
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "\n",
      "Bioconductor version 3.14 (BiocManager 1.30.18), R 4.1.2 (2021-11-01)\n",
      "\n",
      "Warning message:\n",
      "\"package(s) not installed when version(s) same as current; use `force = TRUE` to\n",
      "  re-install: 'IlluminaHumanMethylation450kmanifest'\"\n",
      "Installation paths not writeable, unable to update packages\n",
      "  path: C:/Program Files/R/R-4.1.2/library\n",
      "  packages:\n",
      "    class, cluster, foreign, MASS, Matrix, mgcv, nlme, nnet, rpart, spatial,\n",
      "    survival\n",
      "\n",
      "Old packages: 'checkmate', 'cli', 'colorspace', 'dbplyr', 'dplyr', 'e1071',\n",
      "  'ellipse', 'fansi', 'flexmix', 'GenomicFeatures', 'glue', 'jsonlite',\n",
      "  'languageserver', 'limma', 'locfit', 'magrittr', 'MatchIt', 'matrixStats',\n",
      "  'mclust', 'nloptr', 'openssl', 'parameters', 'plyr', 'processx', 'ps',\n",
      "  'Rcpp', 'restfulr', 'rhdf5', 'rlang', 'roxygen2', 'RSpectra', 'RSQLite',\n",
      "  'Rtsne', 'S4Vectors', 'segmented', 'sp', 'tibble', 'tzdb', 'uuid', 'vctrs',\n",
      "  'XML', 'yaml'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# sources from which this notebook was build:\n",
    "#   R Packages Vignettes: \n",
    "#     https://bioconductor.org/packages/release/bioc/html/GEOquery.html\n",
    "#     https://bioconductor.org/packages/release/bioc/html/minfi.html\n",
    "#     https://bioconductor.org/packages/release/workflows/html/methylationArrayAnalysis.html\n",
    "#   Others:\n",
    "#     https://vbaliga.github.io/verify-that-r-packages-are-installed-and-loaded/\n",
    "    \n",
    "\n",
    "\n",
    "packages = c(\"BiocManager\", \"GEOquery\",  \"minfi\",  \"purrr\",  \"strex\" )\n",
    "\n",
    "package.check <- lapply(\n",
    "    packages,\n",
    "    FUN = function(x) {\n",
    "        if (!require(x, character.only = TRUE)) {\n",
    "            install.packages(x, dependencies = TRUE)\n",
    "            library(x, character.only = TRUE)\n",
    "        }\n",
    "    }\n",
    ")\n",
    "\n",
    "\n",
    "BiocManager::install(\"IlluminaHumanMethylation450kmanifest\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data to load or download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# CHARGE and Kabuki: GSE97362\n",
    "# Sotos : GSE74432\n",
    "\n",
    "GEO = 'GSE97362'\n",
    "DOWNLOAD_GEO = F # F: just load, T: download and load"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Metadata - GEOquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 1 file(s)\n",
      "\n",
      "GSE97362_series_matrix.txt.gz\n",
      "\n",
      "Using locally cached version: ./data//GSE97362_series_matrix.txt.gz\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(GEOquery)\n",
    "\n",
    "metadata <- pData(getGEO(GEO,  GSEMatrix=TRUE, getGPL=FALSE, destdir='./data/')[[1]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 41 × 1 of type chr</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GSM2562708</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>title</th><td>CHD7-10 whole blood CHD7 LOF genomic DNA                                                                                                                                                                                                                                                                                                                                                                    </td></tr>\n",
       "\t<tr><th scope=row>geo_accession</th><td>GSM2562708                                                                                                                                                                                                                                                                                                                                                                                                  </td></tr>\n",
       "\t<tr><th scope=row>status</th><td>Public on May 03 2017                                                                                                                                                                                                                                                                                                                                                                                       </td></tr>\n",
       "\t<tr><th scope=row>submission_date</th><td>Apr 03 2017                                                                                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>last_update_date</th><td>Oct 22 2020                                                                                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>type</th><td>genomic                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><th scope=row>channel_count</th><td>1                                                                                                                                                                                                                                                                                                                                                                                                           </td></tr>\n",
       "\t<tr><th scope=row>source_name_ch1</th><td>whole blood genomic DNA                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><th scope=row>organism_ch1</th><td>Homo sapiens                                                                                                                                                                                                                                                                                                                                                                                                </td></tr>\n",
       "\t<tr><th scope=row>characteristics_ch1</th><td>gender: female                                                                                                                                                                                                                                                                                                                                                                                              </td></tr>\n",
       "\t<tr><th scope=row>characteristics_ch1.1</th><td>age (years): 0.02                                                                                                                                                                                                                                                                                                                                                                                           </td></tr>\n",
       "\t<tr><th scope=row>characteristics_ch1.2</th><td>sample type: CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                      </td></tr>\n",
       "\t<tr><th scope=row>characteristics_ch1.3</th><td>disease state: CHARGE                                                                                                                                                                                                                                                                                                                                                                                       </td></tr>\n",
       "\t<tr><th scope=row>characteristics_ch1.4</th><td>tissue: whole blood                                                                                                                                                                                                                                                                                                                                                                                         </td></tr>\n",
       "\t<tr><th scope=row>molecule_ch1</th><td>genomic DNA                                                                                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>extract_protocol_ch1</th><td>All genomic DNA were extracted from whole blood using standard procedures. One microgram of genomic DNA was sodium bisulfite modified using the Qiagen EZ DNA Methylation kit (Qiagen, Valencia, CA) according to the manufacturerâ€™s protocol. Modified genomic DNA was then processed and analyzed on the Infinium HumanMethylation450 BeadChip from Illumina according to the manufacturerâ€™s protocol.</td></tr>\n",
       "\t<tr><th scope=row>label_ch1</th><td>Cy5 and Cy3                                                                                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>label_protocol_ch1</th><td>Standard Illumina Protocol                                                                                                                                                                                                                                                                                                                                                                                  </td></tr>\n",
       "\t<tr><th scope=row>taxid_ch1</th><td>9606                                                                                                                                                                                                                                                                                                                                                                                                        </td></tr>\n",
       "\t<tr><th scope=row>hyb_protocol</th><td>Bisulphite converted DNA was amplified, fragmented and hybridized to Illumina Infinium Human Methylation450 Beadchip using standard Illumina protocol                                                                                                                                                                                                                                                       </td></tr>\n",
       "\t<tr><th scope=row>scan_protocol</th><td>Using an Illumina HiScan                                                                                                                                                                                                                                                                                                                                                                                    </td></tr>\n",
       "\t<tr><th scope=row>description</th><td>CHD7-10                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><th scope=row>data_processing</th><td>Background subtraction and normalization to internal array controls were performed on raw intensity data using Illumina GenomeStudio v2011.1 Methylation Module 1.9 software. Probes containing single nucleotide polymorphisms (SNPs), cross-reactive probes with sequences aligned to multiple positions with â‰¥ 90% identity, and non-specific chromosome probes were removed.                          </td></tr>\n",
       "\t<tr><th scope=row>platform_id</th><td>GPL13534                                                                                                                                                                                                                                                                                                                                                                                                    </td></tr>\n",
       "\t<tr><th scope=row>contact_name</th><td>Rosanna,,Weksberg                                                                                                                                                                                                                                                                                                                                                                                           </td></tr>\n",
       "\t<tr><th scope=row>contact_laboratory</th><td>Weksberg Lab                                                                                                                                                                                                                                                                                                                                                                                                </td></tr>\n",
       "\t<tr><th scope=row>contact_department</th><td>Genetics and Genome Biology                                                                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>contact_institute</th><td>The Hospital for Sick Children                                                                                                                                                                                                                                                                                                                                                                              </td></tr>\n",
       "\t<tr><th scope=row>contact_address</th><td>555 University Ave.                                                                                                                                                                                                                                                                                                                                                                                         </td></tr>\n",
       "\t<tr><th scope=row>contact_city</th><td>Toronto                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><th scope=row>contact_state</th><td>Ontario                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><th scope=row>contact_zip/postal_code</th><td>M5G 2L3                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>\n",
       "\t<tr><th scope=row>contact_country</th><td>Canada                                                                                                                                                                                                                                                                                                                                                                                                      </td></tr>\n",
       "\t<tr><th scope=row>supplementary_file</th><td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708_9376525052_R01C01_Grn.idat.gz                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>supplementary_file.1</th><td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708_9376525052_R01C01_Red.idat.gz                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "\t<tr><th scope=row>data_row_count</th><td>485577                                                                                                                                                                                                                                                                                                                                                                                                      </td></tr>\n",
       "\t<tr><th scope=row>age (years):ch1</th><td>0.02                                                                                                                                                                                                                                                                                                                                                                                                        </td></tr>\n",
       "\t<tr><th scope=row>disease state:ch1</th><td>CHARGE                                                                                                                                                                                                                                                                                                                                                                                                      </td></tr>\n",
       "\t<tr><th scope=row>gender:ch1</th><td>female                                                                                                                                                                                                                                                                                                                                                                                                      </td></tr>\n",
       "\t<tr><th scope=row>sample type:ch1</th><td>CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                                   </td></tr>\n",
       "\t<tr><th scope=row>tissue:ch1</th><td>whole blood                                                                                                                                                                                                                                                                                                                                                                                                 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 41 × 1 of type chr\n",
       "\\begin{tabular}{r|l}\n",
       "  & GSM2562708\\\\\n",
       "\\hline\n",
       "\ttitle & CHD7-10 whole blood CHD7 LOF genomic DNA                                                                                                                                                                                                                                                                                                                                                                    \\\\\n",
       "\tgeo\\_accession & GSM2562708                                                                                                                                                                                                                                                                                                                                                                                                  \\\\\n",
       "\tstatus & Public on May 03 2017                                                                                                                                                                                                                                                                                                                                                                                       \\\\\n",
       "\tsubmission\\_date & Apr 03 2017                                                                                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\tlast\\_update\\_date & Oct 22 2020                                                                                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\ttype & genomic                                                                                                                                                                                                                                                                                                                                                                                                     \\\\\n",
       "\tchannel\\_count & 1                                                                                                                                                                                                                                                                                                                                                                                                           \\\\\n",
       "\tsource\\_name\\_ch1 & whole blood genomic DNA                                                                                                                                                                                                                                                                                                                                                                                     \\\\\n",
       "\torganism\\_ch1 & Homo sapiens                                                                                                                                                                                                                                                                                                                                                                                                \\\\\n",
       "\tcharacteristics\\_ch1 & gender: female                                                                                                                                                                                                                                                                                                                                                                                              \\\\\n",
       "\tcharacteristics\\_ch1.1 & age (years): 0.02                                                                                                                                                                                                                                                                                                                                                                                           \\\\\n",
       "\tcharacteristics\\_ch1.2 & sample type: CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                      \\\\\n",
       "\tcharacteristics\\_ch1.3 & disease state: CHARGE                                                                                                                                                                                                                                                                                                                                                                                       \\\\\n",
       "\tcharacteristics\\_ch1.4 & tissue: whole blood                                                                                                                                                                                                                                                                                                                                                                                         \\\\\n",
       "\tmolecule\\_ch1 & genomic DNA                                                                                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\textract\\_protocol\\_ch1 & All genomic DNA were extracted from whole blood using standard procedures. One microgram of genomic DNA was sodium bisulfite modified using the Qiagen EZ DNA Methylation kit (Qiagen, Valencia, CA) according to the manufacturerâ€™s protocol. Modified genomic DNA was then processed and analyzed on the Infinium HumanMethylation450 BeadChip from Illumina according to the manufacturerâ€™s protocol.\\\\\n",
       "\tlabel\\_ch1 & Cy5 and Cy3                                                                                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\tlabel\\_protocol\\_ch1 & Standard Illumina Protocol                                                                                                                                                                                                                                                                                                                                                                                  \\\\\n",
       "\ttaxid\\_ch1 & 9606                                                                                                                                                                                                                                                                                                                                                                                                        \\\\\n",
       "\thyb\\_protocol & Bisulphite converted DNA was amplified, fragmented and hybridized to Illumina Infinium Human Methylation450 Beadchip using standard Illumina protocol                                                                                                                                                                                                                                                       \\\\\n",
       "\tscan\\_protocol & Using an Illumina HiScan                                                                                                                                                                                                                                                                                                                                                                                    \\\\\n",
       "\tdescription & CHD7-10                                                                                                                                                                                                                                                                                                                                                                                                     \\\\\n",
       "\tdata\\_processing & Background subtraction and normalization to internal array controls were performed on raw intensity data using Illumina GenomeStudio v2011.1 Methylation Module 1.9 software. Probes containing single nucleotide polymorphisms (SNPs), cross-reactive probes with sequences aligned to multiple positions with â‰¥ 90\\% identity, and non-specific chromosome probes were removed.                          \\\\\n",
       "\tplatform\\_id & GPL13534                                                                                                                                                                                                                                                                                                                                                                                                    \\\\\n",
       "\tcontact\\_name & Rosanna,,Weksberg                                                                                                                                                                                                                                                                                                                                                                                           \\\\\n",
       "\tcontact\\_laboratory & Weksberg Lab                                                                                                                                                                                                                                                                                                                                                                                                \\\\\n",
       "\tcontact\\_department & Genetics and Genome Biology                                                                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\tcontact\\_institute & The Hospital for Sick Children                                                                                                                                                                                                                                                                                                                                                                              \\\\\n",
       "\tcontact\\_address & 555 University Ave.                                                                                                                                                                                                                                                                                                                                                                                         \\\\\n",
       "\tcontact\\_city & Toronto                                                                                                                                                                                                                                                                                                                                                                                                     \\\\\n",
       "\tcontact\\_state & Ontario                                                                                                                                                                                                                                                                                                                                                                                                     \\\\\n",
       "\tcontact\\_zip/postal\\_code & M5G 2L3                                                                                                                                                                                                                                                                                                                                                                                                     \\\\\n",
       "\tcontact\\_country & Canada                                                                                                                                                                                                                                                                                                                                                                                                      \\\\\n",
       "\tsupplementary\\_file & ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708\\_9376525052\\_R01C01\\_Grn.idat.gz                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\tsupplementary\\_file.1 & ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708\\_9376525052\\_R01C01\\_Red.idat.gz                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\tdata\\_row\\_count & 485577                                                                                                                                                                                                                                                                                                                                                                                                      \\\\\n",
       "\tage (years):ch1 & 0.02                                                                                                                                                                                                                                                                                                                                                                                                        \\\\\n",
       "\tdisease state:ch1 & CHARGE                                                                                                                                                                                                                                                                                                                                                                                                      \\\\\n",
       "\tgender:ch1 & female                                                                                                                                                                                                                                                                                                                                                                                                      \\\\\n",
       "\tsample type:ch1 & CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                                   \\\\\n",
       "\ttissue:ch1 & whole blood                                                                                                                                                                                                                                                                                                                                                                                                 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 41 × 1 of type chr\n",
       "\n",
       "| <!--/--> | GSM2562708 |\n",
       "|---|---|\n",
       "| title | CHD7-10 whole blood CHD7 LOF genomic DNA                                                                                                                                                                                                                                                                                                                                                                     |\n",
       "| geo_accession | GSM2562708                                                                                                                                                                                                                                                                                                                                                                                                   |\n",
       "| status | Public on May 03 2017                                                                                                                                                                                                                                                                                                                                                                                        |\n",
       "| submission_date | Apr 03 2017                                                                                                                                                                                                                                                                                                                                                                                                  |\n",
       "| last_update_date | Oct 22 2020                                                                                                                                                                                                                                                                                                                                                                                                  |\n",
       "| type | genomic                                                                                                                                                                                                                                                                                                                                                                                                      |\n",
       "| channel_count | 1                                                                                                                                                                                                                                                                                                                                                                                                            |\n",
       "| source_name_ch1 | whole blood genomic DNA                                                                                                                                                                                                                                                                                                                                                                                      |\n",
       "| organism_ch1 | Homo sapiens                                                                                                                                                                                                                                                                                                                                                                                                 |\n",
       "| characteristics_ch1 | gender: female                                                                                                                                                                                                                                                                                                                                                                                               |\n",
       "| characteristics_ch1.1 | age (years): 0.02                                                                                                                                                                                                                                                                                                                                                                                            |\n",
       "| characteristics_ch1.2 | sample type: CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                       |\n",
       "| characteristics_ch1.3 | disease state: CHARGE                                                                                                                                                                                                                                                                                                                                                                                        |\n",
       "| characteristics_ch1.4 | tissue: whole blood                                                                                                                                                                                                                                                                                                                                                                                          |\n",
       "| molecule_ch1 | genomic DNA                                                                                                                                                                                                                                                                                                                                                                                                  |\n",
       "| extract_protocol_ch1 | All genomic DNA were extracted from whole blood using standard procedures. One microgram of genomic DNA was sodium bisulfite modified using the Qiagen EZ DNA Methylation kit (Qiagen, Valencia, CA) according to the manufacturerâ€™s protocol. Modified genomic DNA was then processed and analyzed on the Infinium HumanMethylation450 BeadChip from Illumina according to the manufacturerâ€™s protocol. |\n",
       "| label_ch1 | Cy5 and Cy3                                                                                                                                                                                                                                                                                                                                                                                                  |\n",
       "| label_protocol_ch1 | Standard Illumina Protocol                                                                                                                                                                                                                                                                                                                                                                                   |\n",
       "| taxid_ch1 | 9606                                                                                                                                                                                                                                                                                                                                                                                                         |\n",
       "| hyb_protocol | Bisulphite converted DNA was amplified, fragmented and hybridized to Illumina Infinium Human Methylation450 Beadchip using standard Illumina protocol                                                                                                                                                                                                                                                        |\n",
       "| scan_protocol | Using an Illumina HiScan                                                                                                                                                                                                                                                                                                                                                                                     |\n",
       "| description | CHD7-10                                                                                                                                                                                                                                                                                                                                                                                                      |\n",
       "| data_processing | Background subtraction and normalization to internal array controls were performed on raw intensity data using Illumina GenomeStudio v2011.1 Methylation Module 1.9 software. Probes containing single nucleotide polymorphisms (SNPs), cross-reactive probes with sequences aligned to multiple positions with â‰¥ 90% identity, and non-specific chromosome probes were removed.                           |\n",
       "| platform_id | GPL13534                                                                                                                                                                                                                                                                                                                                                                                                     |\n",
       "| contact_name | Rosanna,,Weksberg                                                                                                                                                                                                                                                                                                                                                                                            |\n",
       "| contact_laboratory | Weksberg Lab                                                                                                                                                                                                                                                                                                                                                                                                 |\n",
       "| contact_department | Genetics and Genome Biology                                                                                                                                                                                                                                                                                                                                                                                  |\n",
       "| contact_institute | The Hospital for Sick Children                                                                                                                                                                                                                                                                                                                                                                               |\n",
       "| contact_address | 555 University Ave.                                                                                                                                                                                                                                                                                                                                                                                          |\n",
       "| contact_city | Toronto                                                                                                                                                                                                                                                                                                                                                                                                      |\n",
       "| contact_state | Ontario                                                                                                                                                                                                                                                                                                                                                                                                      |\n",
       "| contact_zip/postal_code | M5G 2L3                                                                                                                                                                                                                                                                                                                                                                                                      |\n",
       "| contact_country | Canada                                                                                                                                                                                                                                                                                                                                                                                                       |\n",
       "| supplementary_file | ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708_9376525052_R01C01_Grn.idat.gz                                                                                                                                                                                                                                                                                                  |\n",
       "| supplementary_file.1 | ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708_9376525052_R01C01_Red.idat.gz                                                                                                                                                                                                                                                                                                  |\n",
       "| data_row_count | 485577                                                                                                                                                                                                                                                                                                                                                                                                       |\n",
       "| age (years):ch1 | 0.02                                                                                                                                                                                                                                                                                                                                                                                                         |\n",
       "| disease state:ch1 | CHARGE                                                                                                                                                                                                                                                                                                                                                                                                       |\n",
       "| gender:ch1 | female                                                                                                                                                                                                                                                                                                                                                                                                       |\n",
       "| sample type:ch1 | CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                                    |\n",
       "| tissue:ch1 | whole blood                                                                                                                                                                                                                                                                                                                                                                                                  |\n",
       "\n"
      ],
      "text/plain": [
       "                        GSM2562708                                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "title                   CHD7-10 whole blood CHD7 LOF genomic DNA                                                                                                                                                                                                                                                                                                                                                                    \n",
       "geo_accession           GSM2562708                                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "status                  Public on May 03 2017                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "submission_date         Apr 03 2017                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "last_update_date        Oct 22 2020                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "type                    genomic                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "channel_count           1                                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "source_name_ch1         whole blood genomic DNA                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "organism_ch1            Homo sapiens                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "characteristics_ch1     gender: female                                                                                                                                                                                                                                                                                                                                                                                              \n",
       "characteristics_ch1.1   age (years): 0.02                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "characteristics_ch1.2   sample type: CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                      \n",
       "characteristics_ch1.3   disease state: CHARGE                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "characteristics_ch1.4   tissue: whole blood                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "molecule_ch1            genomic DNA                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "extract_protocol_ch1    All genomic DNA were extracted from whole blood using standard procedures. One microgram of genomic DNA was sodium bisulfite modified using the Qiagen EZ DNA Methylation kit (Qiagen, Valencia, CA) according to the manufacturerâ\\200\\231s protocol. Modified genomic DNA was then processed and analyzed on the Infinium HumanMethylation450 BeadChip from Illumina according to the manufacturerâ\\200\\231s protocol.\n",
       "label_ch1               Cy5 and Cy3                                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "label_protocol_ch1      Standard Illumina Protocol                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "taxid_ch1               9606                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "hyb_protocol            Bisulphite converted DNA was amplified, fragmented and hybridized to Illumina Infinium Human Methylation450 Beadchip using standard Illumina protocol                                                                                                                                                                                                                                                       \n",
       "scan_protocol           Using an Illumina HiScan                                                                                                                                                                                                                                                                                                                                                                                    \n",
       "description             CHD7-10                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "data_processing         Background subtraction and normalization to internal array controls were performed on raw intensity data using Illumina GenomeStudio v2011.1 Methylation Module 1.9 software. Probes containing single nucleotide polymorphisms (SNPs), cross-reactive probes with sequences aligned to multiple positions with â‰¥ 90% identity, and non-specific chromosome probes were removed.                          \n",
       "platform_id             GPL13534                                                                                                                                                                                                                                                                                                                                                                                                    \n",
       "contact_name            Rosanna,,Weksberg                                                                                                                                                                                                                                                                                                                                                                                           \n",
       "contact_laboratory      Weksberg Lab                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "contact_department      Genetics and Genome Biology                                                                                                                                                                                                                                                                                                                                                                                 \n",
       "contact_institute       The Hospital for Sick Children                                                                                                                                                                                                                                                                                                                                                                              \n",
       "contact_address         555 University Ave.                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "contact_city            Toronto                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "contact_state           Ontario                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "contact_zip/postal_code M5G 2L3                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "contact_country         Canada                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "supplementary_file      ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708_9376525052_R01C01_Grn.idat.gz                                                                                                                                                                                                                                                                                                 \n",
       "supplementary_file.1    ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2562nnn/GSM2562708/suppl/GSM2562708_9376525052_R01C01_Red.idat.gz                                                                                                                                                                                                                                                                                                 \n",
       "data_row_count          485577                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "age (years):ch1         0.02                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "disease state:ch1       CHARGE                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "gender:ch1              female                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "sample type:ch1         CHD7 LOF discovery cohort                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "tissue:ch1              whole blood                                                                                                                                                                                                                                                                                                                                                                                                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "t(metadata[10,]) # we just visualize for one subject "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Metadata for GSE97362 ---> DONE"
     ]
    }
   ],
   "source": [
    "library(purrr)\n",
    "library(strex)\n",
    "\n",
    "\n",
    "\n",
    "if (GEO == \"GSE97362\") {\n",
    "    metadata <- metadata[, c(\"geo_accession\", \"description\", \"disease state:ch1\", \"gender:ch1\", \"age (years):ch1\")]\n",
    "    colnames(metadata) <- c(\"geo_accession\", \"description\", \"disease_state\", \"gender\", \"age\")\n",
    "    head(metadata)\n",
    "\n",
    "    # - DISEASE ID\n",
    "    metadata$disease_id <- as.character(map(strsplit(metadata$description, split = \"-\", 1), 1))\n",
    "    metadata[metadata$disease_state == \"Control\",]$disease_id <- \"CTRL\"\n",
    "    head(metadata)\n",
    "\n",
    "\n",
    "    # - AGE CATEGORY\n",
    "    age_round <-round(as.numeric(str_extract_numbers(metadata$age,  decimals = TRUE)))\n",
    "    metadata$age <- age_round\n",
    "\n",
    "        # check for NA's remove them\n",
    "    bad_samples = metadata$geo_accession[is.na(age_round)]\n",
    "    bad_samples\n",
    "    metadata <- metadata[!metadata$geo_accession %in% bad_samples, ]\n",
    "    #age_round[is.na(age_round)] <- mean(age_round, na.rm = TRUE)  # OR replace by mean\n",
    "\n",
    "        # cut by levels\n",
    "    metadata$age_cat <- cut(metadata$age, breaks = c(-1, 14, 24, 64, 150), labels = c('child','youth','adults','senior'))\n",
    "    head(metadata)\n",
    "\n",
    "        # just keep some rows\n",
    "    metadata <- metadata[, c('geo_accession', 'disease_state', 'gender', 'disease_id', 'age_cat', 'age')]\n",
    "    head(metadata)\n",
    "    dim(metadata)\n",
    "    \n",
    "    cat(\"Metadata for GSE97362 ---> DONE\")\n",
    "    \n",
    "    \n",
    "} else if (GEO == \"GSE74432\") {\n",
    "    metadata <- metadata[, c(\"geo_accession\", \"disease state:ch1\", \"gender:ch1\", \"characteristics_ch1.2\")]\n",
    "    colnames(metadata) <-  c(\"geo_accession\", \"disease_state\", \"gender\",\"tissue\")\n",
    "    head(metadata)\n",
    "    \n",
    "    # - KEEP ONLY BLOOD TISSUE\n",
    "    tissue_samples <-  metadata[metadata$tissue == 'tissue: whole blood', ]$geo_accession\n",
    "    metadata <- metadata[tissue_samples, ]\n",
    "    \n",
    "    # - DISEASE ID\n",
    "    metadata$disease_id <- as.character(rep(NA, nrow(metadata)))\n",
    "    metadata[metadata$disease_state == \"Control\",]$disease_id <- \"CTRL\"\n",
    "    metadata[metadata$disease_state == \"Sotos\",]$disease_id <- \"NSD1\"\n",
    "    metadata[metadata$disease_state == \"NSD1 variant\",]$disease_id <- \"NSD1_V\"\n",
    "    metadata <- metadata[!is.na(metadata$disease_id),   ]\n",
    "    head(metadata)\n",
    "    \n",
    "    \n",
    "    metadata <- metadata[,!colnames(metadata) %in% c('tissue')]\n",
    "    cat(\"Metadata for GSE74432 ---> DONE\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>geo_accession</dt><dd>FALSE</dd><dt>disease_state</dt><dd>FALSE</dd><dt>gender</dt><dd>FALSE</dd><dt>disease_id</dt><dd>FALSE</dd><dt>age_cat</dt><dd>FALSE</dd><dt>age</dt><dd>FALSE</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[geo\\textbackslash{}\\_accession] FALSE\n",
       "\\item[disease\\textbackslash{}\\_state] FALSE\n",
       "\\item[gender] FALSE\n",
       "\\item[disease\\textbackslash{}\\_id] FALSE\n",
       "\\item[age\\textbackslash{}\\_cat] FALSE\n",
       "\\item[age] FALSE\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "geo_accession\n",
       ":   FALSEdisease_state\n",
       ":   FALSEgender\n",
       ":   FALSEdisease_id\n",
       ":   FALSEage_cat\n",
       ":   FALSEage\n",
       ":   FALSE\n",
       "\n"
      ],
      "text/plain": [
       "geo_accession disease_state        gender    disease_id       age_cat \n",
       "        FALSE         FALSE         FALSE         FALSE         FALSE \n",
       "          age \n",
       "        FALSE "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# just to be sure we check for NAs\n",
    "\n",
    "apply(metadata, 2, function(x) any(is.na(x)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Raw data - minfi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we load the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n",
      "Warning message in readChar(con, nchars = n):\n",
      "\"truncating string with embedded nuls\"\n"
     ]
    }
   ],
   "source": [
    "library(GEOquery)\n",
    "library(minfi)\n",
    "library(IlluminaHumanMethylation450kmanifest)\n",
    "\n",
    "\n",
    "if (DOWNLOAD_GEO) {\n",
    "    # get RAW_tar file (to download the raw)\n",
    "    getGEOSuppFiles(GEO)\n",
    " \n",
    "\n",
    "    # unatar and unzip everything\n",
    "    untar( tarfile = file.path(parse(text=paste(GEO,'/',GEO,'_RAW.tar',sep=''))), \n",
    "           exdir = file.path(parse(text=paste(GEO,'/idat',sep=''))))\n",
    "\n",
    "    idatFiles <- list.files( file.path(parse(text=paste(GEO,'/idat',sep=''))) , pattern = \".gz$\", full = TRUE)\n",
    "    sapply(idatFiles, gunzip, overwrite = TRUE)\n",
    "}\n",
    "\n",
    "# Load the Methylation Array\n",
    "RGset <- read.metharray.exp( file.path(parse(text=paste(GEO,'/idat',sep=''))) ) #R/G channel"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we reformat the colnames (samples ID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: RGChannelSet \n",
       "dim: 622399 235 \n",
       "metadata(0):\n",
       "assays(2): Green Red\n",
       "rownames(622399): 10600313 10600322 ... 74810490 74810492\n",
       "rowData names(0):\n",
       "colnames(235): GSM2562699 GSM2562700 ... GSM2562934 GSM2562935\n",
       "colData names(0):\n",
       "Annotation\n",
       "  array: IlluminaHumanMethylation450k\n",
       "  annotation: ilmn12.hg19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# see here: https://kasperdanielhansen.github.io/genbioconductor/html/minfi.html\n",
    "\n",
    "colnames(RGset) <- as.character(map(strsplit(colnames(RGset), split = \"_\", 1), 1)) # from GSM2562699_9376525031_R05C01t to GSM2562699\n",
    "RGset\n",
    "# pd <- pData(RGset) \n",
    "# pd # this phenotype data is not usefull here : col of size 0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we remove the bad samples "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: RGChannelSet \n",
       "dim: 622399 234 \n",
       "metadata(0):\n",
       "assays(2): Green Red\n",
       "rownames(622399): 10600313 10600322 ... 74810490 74810492\n",
       "rowData names(0):\n",
       "colnames(234): GSM2562699 GSM2562700 ... GSM2562934 GSM2562935\n",
       "colData names(0):\n",
       "Annotation\n",
       "  array: IlluminaHumanMethylation450k\n",
       "  annotation: ilmn12.hg19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "RGset <- RGset[, colnames(RGset) %in% metadata$geo_accession]\n",
    "RGset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save RGset & metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "save(RGset,metadata,file=file.path( parse(text=paste('RAW_',GEO,'_RGset_metadata.RData',sep='')) ))  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.1.2 (2021-11-01)\n",
       "Platform: x86_64-w64-mingw32/x64 (64-bit)\n",
       "Running under: Windows 10 x64 (build 22000)\n",
       "\n",
       "Matrix products: default\n",
       "\n",
       "locale:\n",
       "[1] LC_COLLATE=English_Belgium.1252  LC_CTYPE=English_Belgium.1252   \n",
       "[3] LC_MONETARY=English_Belgium.1252 LC_NUMERIC=C                    \n",
       "[5] LC_TIME=English_Belgium.1252    \n",
       "\n",
       "attached base packages:\n",
       "[1] parallel  stats4    stats     graphics  grDevices utils     datasets \n",
       "[8] methods   base     \n",
       "\n",
       "other attached packages:\n",
       " [1] IlluminaHumanMethylation450kmanifest_0.4.0\n",
       " [2] strex_1.4.2                               \n",
       " [3] stringr_1.4.0                             \n",
       " [4] purrr_0.3.4                               \n",
       " [5] minfi_1.40.0                              \n",
       " [6] bumphunter_1.36.0                         \n",
       " [7] locfit_1.5-9.4                            \n",
       " [8] iterators_1.0.14                          \n",
       " [9] foreach_1.5.2                             \n",
       "[10] Biostrings_2.62.0                         \n",
       "[11] XVector_0.34.0                            \n",
       "[12] SummarizedExperiment_1.24.0               \n",
       "[13] MatrixGenerics_1.6.0                      \n",
       "[14] matrixStats_0.61.0                        \n",
       "[15] GenomicRanges_1.46.1                      \n",
       "[16] GenomeInfoDb_1.30.1                       \n",
       "[17] IRanges_2.28.0                            \n",
       "[18] S4Vectors_0.32.3                          \n",
       "[19] GEOquery_2.62.2                           \n",
       "[20] Biobase_2.54.0                            \n",
       "[21] BiocGenerics_0.40.0                       \n",
       "[22] BiocManager_1.30.18                       \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       "  [1] rjson_0.2.21              ellipsis_0.3.2           \n",
       "  [3] siggenes_1.68.0           mclust_5.4.9             \n",
       "  [5] IRdisplay_1.1             base64enc_0.1-3          \n",
       "  [7] base64_2.0                bit64_4.0.5              \n",
       "  [9] AnnotationDbi_1.56.2      fansi_1.0.2              \n",
       " [11] xml2_1.3.3                R.methodsS3_1.8.1        \n",
       " [13] codetools_0.2-18          splines_4.1.2            \n",
       " [15] sparseMatrixStats_1.6.0   scrime_1.3.5             \n",
       " [17] cachem_1.0.6              IRkernel_1.3             \n",
       " [19] jsonlite_1.7.3            Rsamtools_2.10.0         \n",
       " [21] annotate_1.72.0           dbplyr_2.1.1             \n",
       " [23] R.oo_1.24.0               png_0.1-7                \n",
       " [25] HDF5Array_1.22.1          readr_2.1.2              \n",
       " [27] compiler_4.1.2            httr_1.4.3               \n",
       " [29] backports_1.4.1           assertthat_0.2.1         \n",
       " [31] Matrix_1.3-4              fastmap_1.1.0            \n",
       " [33] limma_3.50.0              cli_3.1.1                \n",
       " [35] htmltools_0.5.2           prettyunits_1.1.1        \n",
       " [37] tools_4.1.2               glue_1.6.1               \n",
       " [39] GenomeInfoDbData_1.2.7    dplyr_1.0.8              \n",
       " [41] rappdirs_0.3.3            doRNG_1.8.2              \n",
       " [43] Rcpp_1.0.8                vctrs_0.3.8              \n",
       " [45] rhdf5filters_1.6.0        multtest_2.50.0          \n",
       " [47] nlme_3.1-153              preprocessCore_1.56.0    \n",
       " [49] rtracklayer_1.54.0        DelayedMatrixStats_1.16.0\n",
       " [51] lifecycle_1.0.1           restfulr_0.0.13          \n",
       " [53] rngtools_1.5.2            XML_3.99-0.8             \n",
       " [55] beanplot_1.3.1            zlibbioc_1.40.0          \n",
       " [57] MASS_7.3-54               hms_1.1.1                \n",
       " [59] rhdf5_2.38.0              RColorBrewer_1.1-3       \n",
       " [61] yaml_2.2.2                curl_4.3.2               \n",
       " [63] memoise_2.0.1             biomaRt_2.50.3           \n",
       " [65] reshape_0.8.9             stringi_1.7.6            \n",
       " [67] RSQLite_2.2.9             genefilter_1.76.0        \n",
       " [69] BiocIO_1.4.0              checkmate_2.0.0          \n",
       " [71] GenomicFeatures_1.46.4    filelock_1.0.2           \n",
       " [73] BiocParallel_1.28.3       repr_1.1.4               \n",
       " [75] rlang_1.0.1               pkgconfig_2.0.3          \n",
       " [77] bitops_1.0-7              nor1mix_1.3-0            \n",
       " [79] evaluate_0.15             lattice_0.20-45          \n",
       " [81] Rhdf5lib_1.16.0           GenomicAlignments_1.30.0 \n",
       " [83] bit_4.0.4                 tidyselect_1.1.2         \n",
       " [85] plyr_1.8.6                magrittr_2.0.2           \n",
       " [87] R6_2.5.1                  generics_0.1.2           \n",
       " [89] pbdZMQ_0.3-7              DelayedArray_0.20.0      \n",
       " [91] DBI_1.1.2                 pillar_1.7.0             \n",
       " [93] survival_3.2-13           KEGGREST_1.34.0          \n",
       " [95] RCurl_1.98-1.6            tibble_3.1.6             \n",
       " [97] crayon_1.5.1              uuid_1.0-3               \n",
       " [99] utf8_1.2.2                BiocFileCache_2.2.1      \n",
       "[101] tzdb_0.2.0                progress_1.2.2           \n",
       "[103] grid_4.1.2                data.table_1.14.2        \n",
       "[105] blob_1.2.3                digest_0.6.29            \n",
       "[107] xtable_1.8-4              tidyr_1.2.0              \n",
       "[109] R.utils_2.11.0            illuminaio_0.36.0        \n",
       "[111] openssl_1.4.6             quadprog_1.5-8           \n",
       "[113] askpass_1.1              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "ee5cc6fef2d70a7e71ee3826687cbd150f18158e0b1eef11d4f4f92bb920e304"
  },
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
